### ISSUE TITLE
 Add nested field button (link_to_add_nested) add several nested fields instead of 1

### ISSUE THREAD

**Issue by krishnadiamesso** [arielj/vanilla-nested#70]
Hello Ariel,

I am really not sure what I am doing wrong. When I click the add nested field button. Instead of adding 1 item, it adds several. and the number it adds is random. Sometimes it does add 1 item but at times it is 2, another 5, or more. It is really random. There is not really an error showing in the console.

I am running Rails 7.

Also thanks for this gem. I read your post on Ombulabs and saw a youtube video of you with SupeRails by Yaroslav Shmarov.

Here is my code below. I also provided a couple of GIFs that demonstrate the error. Any tips that could point me in the right direction would be really helpful.

```
<div class="px-4 sm:px-6 pt-4 flex justify-end">
    <%= link_to_add_nested(form, :closing_costs, '#closing-costs-items', tag: 'button', partial: 'properties/property_modals/closing_cost_fields', link_classes: "inline-flex items-center rounded-md border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-700 shadow-sm hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-orange-500 focus:ring-offset-2") do %>
      <!--       Heroicon name: plus-->
      <svg class="-ml-1 mr-2 h-5 w-5 text-gray-500" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor">
        <path stroke-linecap="round" stroke-linejoin="round" d="M12 4.5v15m7.5-7.5h-15"/>
      </svg> Add closing cost
    <% end %>
  </div>
...
  <div class="px-4 sm:px-6">
    <div class="pt-6 space-y-6" id="closing-costs-items">
      <%= form.fields_for :closing_costs do |subform| %>
        <%= render partial: 'properties/property_modals/closing_cost_fields', locals: { form: subform } %>
      <% end %>
    </div>
  </div>
```

```
gemfile.rb

gem 'vanilla_nested'
```

```
partial _closing_cost_fields.html.erb
<div class="grid grid-cols-12 gap-4 closing_costs_fields">
  <div class="col-span-5">
    <%= form.text_field :name, class: "block w-full rounded-md border border-gray-300 bg-white text-sm shadow-sm focus:border-orange-500 focus:ring-orange-500", required: true %>
  </div>
  <div class="col-span-5">
    <%= form.number_field :amount, class: "block w-full rounded-md border border-gray-300 bg-white text-sm shadow-sm focus:border-orange-500 focus:ring-orange-500", required: true %>
  </div>
  <div class="col-span-2 flex items-center justify-center">
    <%= link_to_remove_nested(form, link_classes: 'cursor-pointer text-gray-500 hover:text-gray-700', fields_wrapper_selector: '.closing_costs_fields') do %>
      <svg class="h-5 w-5 " xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor">
        <path stroke-linecap="round" stroke-linejoin="round" d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"/>
      </svg>
    <% end %>
  </div>
</div>
```
![2023-03-11_08-42-09 (1)](https://user-images.githubusercontent.com/25694220/224489440-451ef6f0-ee17-43e2-a997-55f888598ea6.gif)

![2023-03-11_09-18-40 (1)](https://user-images.githubusercontent.com/25694220/224489730-6829f940-62e7-42fa-ad23-d6b5518983e5.gif)

**Comment by arielj** [comment]
do you have a repo that I can look at to debug this?

the only thing I can imagine is that something is running multiple times adding the event listener every time, but I can't imagine how's that happening

**Comment by krishnadiamesso** [comment]
Hello Ariel, An update on this.

I ended up implementing the code to add and delete nested fields with straight javascript based on the railscast of Ryan Bates. That worked well for me.

http://railscasts.com/episodes/197-nested-model-form-part-2

I am thinking it has something to do with Hotwire/Rails 7. Because I review the code top to bottom and It seems i had set it up properly. your gem is very nice. very simple and provide a lot of options. and the documentation is straight forward as well.

I have tried cocoon on another branch and cocoon as well was giving me the same behavior of adding several empty nested fields. So the issue is not just your gem.

Thanks for getting back to me. I will go ahead and close the issue.
